src/fragmentarium/ui/SearchForms.tsx

```diff
@@ -0,0 +1,230 @@
+import React, { Component } from 'react'
+import LuckyButton from 'fragmentarium/ui/front-page/LuckyButton'
+import PioneersButton from 'fragmentarium/ui/PioneersButton'
+import { withRouter, RouteComponentProps } from 'react-router-dom'
+import { Button, ButtonToolbar, Col, Form, Popover, Row } from 'react-bootstrap'
+import { stringify } from 'query-string'
+import BibliographySelect from '../../bibliography/ui/BibliographySelect'
+import HelpTrigger from '../../common/HelpTrigger'
+import _ from 'lodash'
+
+interface State {
+  number: string | null | undefined
+  referenceEntry: any
+  pages: string | null | undefined
+  transliteration: string | null | undefined
+  isValid: boolean
+}
+
+type Props = {
+  number: string | null | undefined
+  id: string | null | undefined
+  primaryAuthor: string | null | undefined
+  year: string | null | undefined
+  title: string | null | undefined
+  pages: string | null | undefined
+  transliteration: string | null | undefined
+  fragmentService
+  fragmentSearchService
+  history: History
+} & RouteComponentProps
+
+class SearchForms extends Component<Props, State> {
+  constructor(props) {
+    super(props)
+    this.state = {
+      number: this.props.number || '',
+      referenceEntry: {
+        id: this.props.id || '',
+        title: this.props.title || '',
+        primaryAuthor: this.props.primaryAuthor || '',
+        year: this.props.year || '',
+      },
+      pages: this.props.pages || '',
+      transliteration: this.props.transliteration || '',
+      isValid: this.isValid(this.props.pages || ''),
+    }
+  }
+
+  onChange = (name: string) => (value): void => {
+    this.setState((prevState) => ({ ...prevState, [name]: value }))
+  }
+
+  onChangePages = (value: string): void => {
+    this.setState({ pages: value })
+    this.setState({ isValid: this.isValid(value) })
+  }
+  isValid(pages: string | null | undefined): boolean {
+    return !!(Number(pages) || !pages)
+  }
+
+  onChangeBibliographyReference = (event) => {
+    const newState = { ...this.state }
+    newState.referenceEntry.title = event.cslData.title || ''
+    newState.referenceEntry.id = event.cslData.id || ''
+    newState.referenceEntry.primaryAuthor = event.cslData.author[0].family || ''
+    newState.referenceEntry.year =
+      event.cslData.issued['date-parts'][0][0] || ''
```

Comment:
Do not mutate state. Use [Immer](https://immerjs.github.io/immer/docs/introduction).